llama_stack/apis/scoring/scoring.py

```diff
@@ -18,13 +18,14 @@
 
 @json_schema_type
 class ScoreBatchResponse(BaseModel):
-    dataset_id: str
+    dataset_id: Optional[str] = None
+    results: Dict[str, ScoringResult]
 
 
 @json_schema_type
 class ScoreResponse(BaseModel):
     # each key in the dict is a scoring function name
-    results: List[Dict[str, ScoringResult]]
+    results: Dict[str, ScoringResult]
```

Comment:
why this change? wouldn't you have one dict per one input row?

Reply:
I had made it s.t. we perform score aggregation (e.g. avg, sum) etc as part of scorer. We can add it back to deligate the aggregation part to the client.

Reply:
Ok, I think aggregation on client might be better, gives a chance for analyzing each example's scores.

Reply:
Updated scoring response output

```
results={
  'equality': ScoringResult(score_rows=[
    {
      'score': 0.0
    },
    {
      'score': 1.0
    },
    {
      'score': 1.0
    },
    {
      'score': 0.0
    }
  ],
  aggregated_results={
    'accuracy': 0.5,
    'num_correct': 2.0,
    'num_total': 4
  })
}
```